src/classes/ADDR_Contact_TDTM.cls

```diff
@@ -29,439 +29,597 @@
+                if(originalContactIDs.contains(child.ID)) {
+                    originalContact = child;
+                    break;
+                }
+            }
+            //Copy Address info to parent and siblings, if it's different
+            if(originalContact != null && !ADDR_Addresses_UTIL.isContactAccAddrInfoSame(originalContact, acc)) {
+	            
+	            //Copy address info to parent Account
+	            ADDR_Addresses_UTIL.copyAddressStdSObj(originalContact, 'Mailing', acc, 'Billing');
+	            objectsToUpdate.add(acc);
+	            
+	            //Copy address info to each sibling - we need to exclude the one the trigger is running on!
+	            for(Contact contact : acc.Contacts) {
+	                if(contact.ID != originalContact.ID && !contact.is_Address_Override__c) {
+	                   contact.Primary_Address_Type__c = originalContact.Primary_Address_Type__c;
+	                   ADDR_Addresses_UTIL.copyAddressStdSObj(originalContact, 'Mailing', contact, 'Mailing');
+	                   objectsToUpdate.add(contact);
+	                }
+	            }
+	        //If the address info is the same, just update the Current_Address__c field in the original contact
+            } else if(originalContact != null && ADDR_Addresses_UTIL.isContactAccAddrInfoSame(originalContact, acc)) {
+                originalContact.Current_Address__c = acc.Current_Address__c;
+                objectsToUpdate.add(originalContact);
+            }
+        }
+        //explicitly doing the update, because if we put the records in DmlWrapper we get a "duplicate id in list" error due to
+        //the method updateAccountPrimaryContact in ACCT_IndividualAccounts_TDTM also putting them there.
+        update objectsToUpdate;   
     }
     
-    /*******************************************************************************************************
-    * @description returns whether the contact's mailing address is empty
-    * @param con1 a Contact
-    * @return boolean. true if the Mailing Address fields are all empty 
-    ********************************************************************************************************/
-    private boolean isContactAddressEmpty(Contact con1) {
-        boolean isEmpty = 
-            con1.MailingStreet == null &&
-            con1.MailingCity == null &&
-            con1.MailingState == null &&
-            con1.MailingPostalCode == null &&
-            // note that we decided to ignore country so that a default value won't create unnecessary address objects
-            (con1.MailingCountry == null || ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled) &&
-            // only test StateCode if picklists enabled.
-            (!ADDR_Addresses_TDTM.isStateCountryPicklistsEnabled || con1.get('MailingStateCode') == null);
-        return isEmpty;   
-           
+    private void propagateAddrInfoDelete(List<Contact> contactsAddrPropagate, DmlWrapper dmlWrapper) {
```

Comment:
general comment.  all of the private methods in your pull request would still greatly benefit from having apexdoc method headers, so one can know the intent of the method.